src/App.tsx

```diff
@@ -1,12 +1,79 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import 'bulma/css/bulma.min.css';
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { LoadingError } from './components/LoadingError/LoadingError';
+import { Loader } from './components/Loader/Loader';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[] | [];
+  isLoading: boolean;
+  hasLoadingError: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state:State = {
+    goods: [],
+    isLoading: false,
+    hasLoadingError: false,
+  };
+
+  loadData = async (getData: () => Promise<Good[]>) => {
+    this.setState({
+      isLoading: true,
+    });
+
+    try {
+      const goods = await getData();
+
+      this.setState({
+        goods,
+        isLoading: false,
+      });
+    } catch {
+      this.setState({
+        hasLoadingError: true,
+      });
```

Comment:
If the error occurs, `isLoading` won't become false, and you know what happens next)
Set `isLoading` to false in the `final` clause, this way you can even omit it in the `try` clause